central/imageintegration/datastore/singleton.go

```diff
@@ -22,16 +23,26 @@ var (
 	dataStore DataStore
 )
 
-func initializeDefaultIntegrations(storage store.Store) {
+func initializeIntegrations(storage store.Store) {
 	ctx := sac.WithGlobalAccessScopeChecker(context.Background(), sac.AllowAllAccessScopeChecker())
 	iis, err := storage.GetAll(ctx)
 	utils.CrashOnError(err)
+	// If we are starting from scratch in online-mode, add the default image integrations.
 	if !env.OfflineModeEnv.BooleanSetting() && len(iis) == 0 {
 		// Add default integrations
 		for _, ii := range store.DefaultImageIntegrations {
 			utils.Must(storage.Upsert(ctx, ii))
 		}
 	}
+
+	// If the feature flag is disabled, remove all "sourced" autogenerated registries from the datastore.
+	if !features.SourcedAutogeneratedIntegrations.Enabled() {
+		for _, ii := range iis {
+			if ii.GetAutogenerated() && ii.GetSource() != nil {
+				utils.Must(storage.Delete(ctx, ii.GetId()))
+			}
+		}
```

Comment:
Let's log the successful delete. 

```suggestion
		}
		if len(iis) > 0 {
                   log.Infof("Successfully deleted %d image integration", len(is))
                }
```

Reply:
very useful for debugging. good idea

Reply:
I wouldn't do Infof. I'd do a Debug. We don't want 11620 or so log messages.

Reply:
I put it outside the loop, so it'l only be logged once (I hope)

Reply:
Ack yes. Time to call it a day.